### PR TITLE
feat: allow config auto gen options for dir

### PR DESCRIPTION
**Pull request by aladdin-add**

### CODE REVIEW

File: test/conf/options.ts [hunk 1]
```diff
@@ -0,0 +1,10 @@
+/**
+ * @fileoverview the default options for different dirs
+ * @author 唯然<weiran.zsd@outlook.com>
+ */
+
+// TODO: add your
+export const defaultOptions = [
+  { dir: 'test/__snapshot__/parser/', parserOptions: {}, printerOptions: {} },
```
Comment (aladdin-add) [inline test/conf/options.ts:8]: you can config your default options here.

File: test/runner/autogenerate.ts [hunk 2]
```diff
@@ -113,16 +114,25 @@ export async function autogen(files: string[], conservative: boolean) {
...
+        };
         // immediately generate a test case for it, as well
         writeFileSync(
           testFile,
           `# Auto-generated test cases ( Kataw )
 - Regenerated: ${new Date().toISOString().slice(0, 10)}
-- From: ${obj.file.slice(obj.file.indexOf('kataw')).replace('\\', '/')}
-- Path: ${caseDir.slice(caseDir.indexOf('kataw')).replace('\\', '/')}
+- From: ${obj.file.slice(obj.file.indexOf('kataw')).replace(/\\/g, '/')}
+- Path: ${caseDir.slice(caseDir.indexOf('kataw')).replace(/\\/g, '/')}
```
Comment (aladdin-add) [inline test/runner/autogenerate.ts:127]: it fixes the wrong path in the md files( windows => unix-style).

File: test/runner/autogenerate.ts [hunk 3]
```diff
@@ -113,16 +114,25 @@ export async function autogen(files: string[], conservative: boolean) {
...
 > :: test: ${title.split('\n').join('\n>          ')}
 > :: case: ${c.split('\n').join('\n>          ')}
+## Options
+
+###
+\`\`\`\`\`js
+${JSON.stringify(options.parserOptions)}
```
Comment (aladdin-add) [inline test/runner/autogenerate.ts:134]: atm, it only writes the parser options, but you can add other options(e.g. printer options) here.

File: test/conf/options.ts (hunk 1 above)
Comment (KFlash) [reply test/conf/options.ts:8]: can you add 'allowTypes' to this test cases and regenerate all the snapshots?

https://github.com/kataw/kataw/tree/main/test/__snapshot__/parser/declarations/type/arrow-function-type

**Review by KFlash — APPROVED** [review]

File: .eslintrc.js [hunk 4]
```diff
@@ -5,7 +5,7 @@ module.exports = {
     {
       files: ['**/*.js'],
       extends: ['eslint:recommended', 'plugin:node/recommended'],
-      parserOptions: { ecmaVersion: 10 }
+      parserOptions: { ecmaVersion: 10, sourceType: "module" }
```
Comment (KFlash) [inline .eslintrc.js:8]: witch parser are you using here?

Comment (aladdin-add) [reply .eslintrc.js:8]: typescript-eslint

**Comment by KFlash** [comment]
The tests are missing. The 'output'.

File: test/conf/options.ts (hunk 1 above)
Comment (KFlash) [reply test/conf/options.ts]: Did you fix this?

Comment (aladdin-add) [reply test/conf/options.ts]: https://github.com/kataw/kataw/issues/102

Comment (KFlash) [reply test/conf/options.ts]: Give me the exact test. I can't reproduce this

**Comment by KFlash** [comment]
@aladdin-add I doubt you should get any issues with 'ObjectType' now. It works just fine in recovery mode, and it's no longer using any lookaheads so better control on the 'SyntaxKind'.

**Comment by KFlash** [comment]
Just in case. I enabled 'types' in the online demo.

https://kataw.github.io/kataw/kataw_ast/

**Comment by KFlash** [comment]
A few incoming tests. Can you regenerate again?

**Pull request merged**